### ISSUE TITLE
这个模型能不能把导出模型代码补上(models/PaddleNLP/PaddleTextGEN/seq2seq/）

### ISSUE THREAD

**Issue by aixier** [PaddlePaddle/models#4149]
https://github.com/PaddlePaddle/models/tree/c35c10a8fdd69039cc79c76f19cee87d04bc400b/PaddleNLP/PaddleTextGEN/seq2seq

**Comment by 1024er** [comment]
> https://github.com/PaddlePaddle/models/tree/c35c10a8fdd69039cc79c76f19cee87d04bc400b/PaddleNLP/PaddleTextGEN/seq2seq

好的，我尽快修复这个问题

**Comment by lixuan1** [comment]
@1024er 希望尽快修复呀，感谢🙏。

**Comment by lixuan1** [comment]
@1024er [models/PaddleNLP/PaddleTextGEN/seq2seq/](https://github.com/PaddlePaddle/models/tree/develop/PaddleNLP/PaddleTextGEN/seq2seq)你好这个模型的导出模型的代码什么时候可以补充上呀？

**Comment by lixuan1** [comment]
@1024er @Superjomn @reyoung 你们好，我想问一下，该模型中，计算的ppl值，在训练集，验证集和测试集都有计算的这个值，这是perplexity的计算吗？但网上查到的ppl公式不是根据ppl = np.exp(total_loss / word_count)计算的呀，所以这里的ppl是什么度量的计算呢？请求解惑，感谢。具体相关代码如下：

`def eval(data, epoch_id=0):
        eval_data_iter = reader.get_data_iter(data, batch_size, mode='eval')
        total_loss = 0.0
        word_count = 0.0
        for batch_id, batch in enumerate(eval_data_iter):
            input_data_feed, word_num = prepare_input(
                batch, epoch_id, with_lr=False)
            fetch_outs = exe.run(inference_program,
                                 feed=input_data_feed,
                                 fetch_list=[loss.name],
                                 use_program_cache=False)

            cost_train = np.array(fetch_outs[0])

            total_loss += cost_train * batch_size
            word_count += word_num

        ppl = np.exp(total_loss / word_count)

        return ppl`

**Comment by lixuan1** [comment]
@1024er @Superjomn @reyoung 你们好，我想问一下，该模型中，计算的ppl值，在训练集，验证集和测试集都有计算的这个值，这是perplexity的计算吗？但网上查到的ppl公式不是根据ppl = np.exp(total_loss / word_count)计算的呀，所以这里的ppl是什么度量的计算呢？请求解惑，感谢。具体相关代码如下：
`
def eval(data, epoch_id=0):
        eval_data_iter = reader.get_data_iter(data, batch_size, mode='eval')
        total_loss = 0.0
        word_count = 0.0
        for batch_id, batch in enumerate(eval_data_iter):
            input_data_feed, word_num = prepare_input(
                batch, epoch_id, with_lr=False)
            fetch_outs = exe.run(inference_program,
                                 feed=input_data_feed,
                                 fetch_list=[loss.name],
                                 use_program_cache=False)

            cost_train = np.array(fetch_outs[0])

            total_loss += cost_train * batch_size
            word_count += word_num

        ppl = np.exp(total_loss / word_count)

        return ppl
`

**Comment by lixuan1** [comment]
@1024er @Superjomn @reyoung 你们好，我想问一下，该模型中，计算的ppl值，在训练集，验证集和测试集都有计算的这个值，这是perplexity的计算吗？但网上查到的ppl公式不是根据ppl = np.exp(total_loss / word_count)计算的呀，所以这里的ppl是什么度量的计算呢？请求解惑，感谢。具体相关代码如下：
`def eval(data, epoch_id=0):
        eval_data_iter = reader.get_data_iter(data, batch_size, mode='eval')
        total_loss = 0.0
        word_count = 0.0
        for batch_id, batch in enumerate(eval_data_iter):
            input_data_feed, word_num = prepare_input(
                batch, epoch_id, with_lr=False)
            fetch_outs = exe.run(inference_program,
                                 feed=input_data_feed,
                                 fetch_list=[loss.name],
                                 use_program_cache=False)

            cost_train = np.array(fetch_outs[0])

            total_loss += cost_train * batch_size
            word_count += word_num

        ppl = np.exp(total_loss / word_count)

	return ppl`

**Comment by lixuan1** [comment]
@1024er @Superjomn @reyoung 你们好，我想问一下，该模型中，计算的ppl值，在训练集，验证集和测试集都有计算的这个值，这是perplexity的计算吗？但网上查到的ppl公式不是根据ppl = np.exp(total_loss / word_count)计算的呀，所以这里的ppl是什么度量的计算呢？请求解惑，感谢。具体相关代码如下：
```
def eval(data, epoch_id=0):
        eval_data_iter = reader.get_data_iter(data, batch_size, mode='eval')
        total_loss = 0.0
        word_count = 0.0
        for batch_id, batch in enumerate(eval_data_iter):
            input_data_feed, word_num = prepare_input(
                batch, epoch_id, with_lr=False)
            fetch_outs = exe.run(inference_program,
                                 feed=input_data_feed,
                                 fetch_list=[loss.name],
                                 use_program_cache=False)

            cost_train = np.array(fetch_outs[0])

            total_loss += cost_train * batch_size
            word_count += word_num

        ppl = np.exp(total_loss / word_count)

	return ppl
```

**Comment by 1024er** [comment]
> @1024er @Superjomn @reyoung 你们好，我想问一下，该模型中，计算的ppl值，在训练集，验证集和测试集都有计算的这个值，这是perplexity的计算吗？但网上查到的ppl公式不是根据ppl = np.exp(total_loss / word_count)计算的呀，所以这里的ppl是什么度量的计算呢？请求解惑，感谢。具体相关代码如下：
> 
> ```
> def eval(data, epoch_id=0):
>         eval_data_iter = reader.get_data_iter(data, batch_size, mode='eval')
>         total_loss = 0.0
>         word_count = 0.0
>         for batch_id, batch in enumerate(eval_data_iter):
>             input_data_feed, word_num = prepare_input(
>                 batch, epoch_id, with_lr=False)
>             fetch_outs = exe.run(inference_program,
>                                  feed=input_data_feed,
>                                  fetch_list=[loss.name],
>                                  use_program_cache=False)
> 
>             cost_train = np.array(fetch_outs[0])
> 
>             total_loss += cost_train * batch_size
>             word_count += word_num
> 
>         ppl = np.exp(total_loss / word_count)
> 
> 	return ppl
> ```
这里的ppl计算方式是参照texar里面的公式来的，和网上的ppl公式只是实现上略有不同。

**Comment by lixuan1** [comment]
@1024er 你好，有关该模型的导出模型的代码，什么时候可以补充上呢
